src/ORM/Query.php

```diff
@@ -669,6 +670,8 @@ public function cleanCopy()
         $clone->offset(null);
         $clone->mapReduce(null, null, true);
         $clone->formatResults(null, true);
+        $clone->selectTypeMap(new TypeMap());
```

Comment:
Should this be done in `__clone()` I don't know if the typemap should be shared between query clones.